core/src/main/java/org/apache/carbondata/core/datamap/DataMapStoreManager.java

```diff
@@ -513,7 +513,17 @@ public void refreshSegmentCacheIfRequired(CarbonTable carbonTable,
    * @param identifier Table identifier
    */
   public void clearDataMaps(AbsoluteTableIdentifier identifier) {
-    clearDataMaps(identifier, true);
+    CarbonTable carbonTable = getCarbonTable(identifier);
+    boolean launchJob = false;
+    try {
+      // launchJob will be true if either the table has a CGDatamap or index server is enabled for
+      // the specified table.
+      launchJob = new DataMapChooser(carbonTable).hasCGDataMap() || CarbonProperties.getInstance()
+          .isDistributedPruningEnabled(identifier.getDatabaseName(), identifier.getTableName());
+    } catch (IOException e) {
+      LOGGER.warn("Unable to launch job to clear datamaps.", e);
+    }
+    clearDataMaps(identifier, launchJob);
```

Comment:
`launchJob` parameter name is not easy to understand, can you change it to 
`cleareDataMapCache(identifier, clearInAllWorkers)` to indicate that it will clear the datamap cache in all workers

Reply:
done